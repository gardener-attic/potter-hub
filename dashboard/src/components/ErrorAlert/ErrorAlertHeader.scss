.error__icon {
  display: inline-block;
  background: #c7c9c8;
  vertical-align: bottom;
  border-radius: 50%;
  min-height: 30px;
  min-width: 30px;
}

.error__icon > svg {
  margin: auto;
  left: 3px;
  top: 3px;
  color: #ffffff;
  position: relative;
}
