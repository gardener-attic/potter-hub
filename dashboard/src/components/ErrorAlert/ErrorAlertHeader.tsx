import * as React from "react";
import { AlertTriangle } from "react-feather";

import "./ErrorAlertHeader.css";

interface IErrorHeaderProps {
  children: string | JSX.Element | Array<string | JSX.Element>;
  icon?: React.Component;
}

class ErrorPageHeader extends React.Component<IErrorHeaderProps> {
  public render() {
    const { children } = this.props;
    const Icon = this.props.icon || AlertTriangle;
    return (
      <header>
        <div className="margin-b-big">
          <h5 className="type-regular">
            <span className="error__icon margin-r-small">
              <Icon />
            </span>
            {children}
          </h5>
        </div>
      </header>
    );
  }
}

export default ErrorPageHeader;
