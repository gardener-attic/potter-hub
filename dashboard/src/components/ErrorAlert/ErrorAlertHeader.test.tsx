import { shallow } from "enzyme";
import * as React from "react";

import { AlertTriangle, X } from "react-feather";
import ErrorAlertHeader from "./ErrorAlertHeader";

it("renders the heading passed to it", () => {
  const wrapper = shallow(<ErrorAlertHeader>test</ErrorAlertHeader>);
  expect(wrapper.text()).toContain("test");
  expect(wrapper).toMatchSnapshot();
});

it("renders the AlertTriangle icon by default", () => {
  const wrapper = shallow(<ErrorAlertHeader>test</ErrorAlertHeader>);
  expect(wrapper.find(".error__icon").contains(<AlertTriangle />)).toBe(true);
});

it("renders the icon passed in by prop", () => {
  const wrapper = shallow(<ErrorAlertHeader icon={X}>test</ErrorAlertHeader>);
  expect(wrapper.find(".error__icon").contains(<AlertTriangle />)).toBe(false);
  expect(wrapper.find(".error__icon").contains(<X />)).toBe(true);
});
