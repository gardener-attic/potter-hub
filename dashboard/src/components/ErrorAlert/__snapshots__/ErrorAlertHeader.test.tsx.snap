// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the heading passed to it 1`] = `
<header>
  <div
    className="margin-b-big"
  >
    <h5
      className="type-regular"
    >
      <span
        className="error__icon margin-r-small"
      >
        <AlertTriangle
          color="currentColor"
          size="24"
        />
      </span>
      test
    </h5>
  </div>
</header>
`;
